pkg/kapp/diff/change_set_with_versioned_rs.go

```diff
@@ -76,6 +86,72 @@ func (d ChangeSetWithVersionedRs) Calculate() ([]Change, error) {
 	return allChanges, nil
 }
 
+func (d ChangeSetWithVersionedRs) checkForMaxDurationAnn(existingRsGrouped map[string][]ctlres.Resource,
+	existingNonVerRsGrouped map[string]ctlres.Resource) error {
+
+	for _, res := range d.newRs {
+		if val, found := res.Annotations()[maxDurationAnnKey]; found {
+
+			// Validate value of annotation `kapp.k14s.io/max-duration`.
+			duration, err := time.ParseDuration(val)
+			if err != nil {
+				return fmt.Errorf("%s for resource: %s", err.Error(), res.Description())
+			}
```

Comment:
nit: remove the newline before the comment
I don't think we need this comment, instead let's try to make the error a bit more clear, so that it's easier to understand for both users and readers.

Maybe something like 
```go
return fmt.Errorf("Unable to parse max-duration for resource %s": %s, res.Description(), err.Error())
```